Update to Linux install/Miscellanous (#19)
AM2R-Community-Developers/DistributionCenter: docs/faq.md

#### What are the minimum specifications to play AM2R on a PC?
An AMD E-300 APU with Radeon HD Graphics and 3GB of RAM should give you fullspeed for the whole game assuming you play on 4:3 and have custom palettes disabled, with some slight stuttering for X Parasite morphing.  

If you have perfomance problems on Windows, be sure to check out the question "[AM2R runs slow on Windows!](https://am2r-community-developers.github.io/DistributionCenter/faq.html#am2r-runs-slow-on-windows-way-slower-than-previous-versions)".  

> 

Reviewer: Personally, not a fan of the newlines.

Author: Feel free to remove

Reviewer: Takes too much effort on my side to go to your fork, clone, edit, commit, and push. 
Can you do that instead.?

Author: Will do

Author: Done

Reviewer: Then resolve the discussion please :)

These specs will provide full-speed gameplay almost everywhere, with some slowdowns in non-critial areas with a large amount of objects, such as the main caverns of Area 2, Area 3, and Area 5. Weaker GPUs may slow down in Fusion difficulty during X Parasite morphing.

#### Where can I stay up-to-date on AM2R?  
- [Reddit](https://reddit.com/r/AM2R)
- [Discord](https://discord.gg/YTQnkAJ) and [Matrix](https://matrix.to/#/#am2r:matrix.org). Discord and Matrix are bridged together.
- [Community Updates YouTube Channel](https://youtube.com/c/am2rcommunityupdates)
- [Legacy AM2R blog](https://metroid2remake.blogspot.com/) (No community updates will be present here)

#### Can I play AM2R on a Mac?  

> Macs are not officially supported. There is however, an unofficial port available here: <https://github.com/Miepee/AM2R-Autopatcher-Mac>

Reviewer: Whitespace at the end is missing, which breaks formatting of the next paragraph.

Author: Fixed and reviewed in preview.

These specs will provide full-speed gameplay almost everywhere, with some slowdowns in non-critial areas with a large amount of objects, such as the main caverns of Area 2, Area 3, and Area 5. Weaker GPUs may slow down in Fusion difficulty during X Parasite morphing.

#### Where can I stay up-to-date on AM2R?  

> - [Reddit](https://reddit.com/r/AM2R) <br> <!--<br> needed for formatting-->

Reviewer: What formatting is `<br>` making work here? The newlines / list?

Author: I thought it was, but it doesn't seem to be anymore.

> - [Discord](https://discord.gg/YTQnkAJ) and [Matrix](https://matrix.to/#/#am2r:matrix.org) (Connected via a cross-platform bridge) <br>

Reviewer: ```suggestion
- [Discord](https://discord.gg/YTQnkAJ) and [Matrix](https://matrix.to/#/#am2r:matrix.org). Discord and Matrix are bridged together. <br>
```

> - [Community Updates Youtube Channel](https://youtube.com/c/am2rcommunityupdates) <br>

Reviewer: *YouTube

Author: done

> - [Legacy AM2R blog](https://metroid2remake.blogspot.com/) (No community updates will be present here)

Reviewer: I personally am not much of a fan to call it "legacy blog" considering Doc is still posting there from time to time. Maybe call it `DoctorM64's AM2R blog` instead?  
Also, Doc did post a few times about the community updates, so I'd also remove that part in the brackets.

Author: I added the `legacy` tag as it may confuse users who try to find the community updates, then only see the DMCA/Slippy. `Doctor M64's AM2R blog` is ok, as long as it is clarified to not be a place to find community updates.

Reviewer: Some more thoughts:
- would be good to bring the bit.ly link back, as it probably gives doc a few bucks here and there
- Docs link was at the very top as a thing out of respect, since he was (one of) the original creators for AM2R
- Clarification can probably be done via a `-` between link and description. Can then also be done for the other lists as consistency, since most of them would also benefit from a small description.

Author: My thoughts:
* I had no idea the bit.ly made money, had I known that, it would not be changed.
* Very understandable, it should be put back on top
* I'll comment my link descriptions in another comment for you to decide on

Author: DoctorM64's AM2R blog - Archive of the original development directly from the original creator. **No community updates or support will be found here**

Reddit - Subreddit of AM2R discussion, support can be found here, but mainly gameplay videos and news are posted.

Discord/Matrix - Server of AM2R, will help with installation and discussion of AM2R. Discord and Matrix are bridged together.

Community Updates YouTube - Video tutorials, announcements, and gameplay are shown in the updates YouTube channel

Reviewer: > Archive of the original development directly from the original creator. No community updates or support will be found here

Several things I don't like here:
- It's not really an archive per se
- DoctorM64 is not the sole original creator, but one of many
- The boldness of the last sentence
- Currently makes it seem like there's not really much of importance there

I'd also significantly shorten the descriptions:

```diff
- Subreddit of AM2R discussion, support can be found here, but mainly gameplay videos and news are posted.
+ The official AM2R Subreddit.
```

```diff
- Discord/Matrix - Server of AM2R, will help with installation and discussion of AM2R. Discord and Matrix are bridged together.
+ Discord/Matrix - The official AM2R Server/Space, both of which are bridged together.
```

```diff
- Community Updates YouTube - Video tutorials, announcements, and gameplay are shown in the updates YouTube channel
+ Community Updates YouTube - The official AM2R Community Updates YouTube channel.
```

Reviewer: Oh, also I double checked, and turns out that the bitly doesn't make money. So feel free to revert that to the original link in order to avoid a redirect.

Author: Your descriptions are helpful for the communication platforms, I will update them. On the topic of the AM2R blog, I agree that it is not an archive and still updated. On the other hand however, users who go to here will not get any help for community updates and may spam the comments of his recent slippy post. I agree that we should honor the original AM2R team for their work, but it can't be ignored that users will receive no help from the blog, and may even confuse skippy for AM2R community updates. How do these options sound? 
`Original AM2R blog by DoctorM64 and AM2R development team, not associated with current development`
or
`Original AM2R blog by DoctorM64 and AM2R development team, no downloads for AM2R launcher due to DMCA`

I feel they credit all individuals of the AM2R development team while clarifying they are not apart of current development, but there is room for improvement.

Reviewer: Am more a fan of the first one, would change it slightly to `Original AM2R blog by DoctorM64. DoctorM64 and the rest of the  original AM2R development team have no association with the Community Updates`.
Oh, and don't forget to move it to the top please.

Author: Moved to the top and added the updates, formatting does not appear to brake, and the links all seem to work out now. I'll keep this open if there is anything else you want to change.

Reviewer: Looks fine by me, feel free to resolve.

#### Can I play AM2R on a Mac?  

> Macs are not officially supported. There is however, an unofficial port available here: <https://github.com/Miepee/AM2R-Autopatcher-Mac>. Bootcamp will also run in older intel-based macs, giving you full native performance while on the Windows version.

Reviewer: A) If bootcamp should be mentioned, itd be in the lower paragraph.
B) you got rid of the double space, which breaks the paragraph
C) Am personally not much of a fan of mentioning bootcamp, as it's not really playing on MacOS.

Author: Bootcamp is native hardware and will most likely run better than wine/Crossover. I'd just remove bootcamp altogether as it is being phased out on ARM MacOS.

Reviewer: Okay, I'm dumb and didn't see that BootCamp is currently already mentioned.  
> I'd just remove bootcamp altogether as it is being phased out on ARM MacOS.

You underestimate how many people are still out there (And we get requests from) with old mac devices. Don't think it'd be a good idea to remove that then. But a slight reword is likely still in order:
```
Macs are not officially supported. There is however, an unofficial port available here: <https://github.com/Miepee/AM2R-Autopatcher-Mac>

If you’re unable to use that, M1 and x64-based Macs are reportedly able to run AM2R via Crossover, and for older Macs an x86-based system may be able to run AM2R via Wine/Wineskin or through a VM. Keep in mind however, that those solutions will be significantly slower than running it natively.  
Alternatively, you can also use Bootcamp on x64-based Macs in order to run the Windows version of AM2R.
```

Author: I still use a 2012 mac device, don't worry, I just didn't want bootcamp listed as worse than native.

Author: Implemented rewrite

- Install a Linux distro onto your Chromebook, and use the native Linux version. This will lead to better performance, should you struggle with the option above, but only works with select Chromebooks. See this for more details: <https://wiki.galliumos.org/Hardware_Compatibility>

#### Can I play AM2R on the Switch/PlayStation Vita?  

> The Switch and PlayStation Vita are not supported platforms. Ports of the Community Updates for these platforms have been made by third parties, but they were created with pirated development tools and violated SDK NDAs; as such every support question regarding them will be deleted, and may result in a ban from all communication channels.

Reviewer: I'd use "from our communication channels" instead of all.

Author: I agree

Author: done

#### GameMaker: Studio 1.4 has a PlayStation Vita export module. Why can't you make a build?  
That module was discontinued before any of the developers on our team could legitimately obtain a copy. Somebody else who already has the module could export the game, but we can't officially support it as we don't have firsthand access and are unsure of the legal concerns surrounding the Sony developer agreement.

#### Can I play AM2R on [third-party handheld emulation device]?  

> If it is running Android or x86-based Linux, you can probably install the respective version of AM2R. However, these devices are not officially supported as they often use highly specialized versions of Android and Linux. 

Reviewer: Unnecessary whitespace.

Author: I fixed it, not sure how you saw that one

Reviewer: Hasn't been fixed, unnecessary whitespace is still existing.

Author: Must have been a mistake on my part, correct fix applied.

However, there is a community made tool that allows you to run Android Games on ARM Linux, which you can check out here: <https://github.com/JohnnyonFlame/droidports>  
A guide on how to use it can be found in this Reddit post, along with some answered FAQ in the comments: <https://www.reddit.com/r/AM2R/comments/pgp2x7/am2r_raspberry_pi_take_4/>

#### How is AM2R still being updated? Didn't the developers get a DMCA?  
After the DMCA back in 2016 the original team ceased development. However, a talented programmer named YellowAfterlife reconstructed the source code in about 4 months. This source was eventually released to the public, so a few members of the community took it upon themselves to release a "community patch" to fix some major bugs in the game. Thanks to scope creep and too much free time, one update became bug fix patches, then texture changes, until eventually updates began adding new content planned by the original devs before the DMCA. Features like New Game+ and Random Game+ modes were added, along with the planned secret ending, Fusion Difficulty, and even widescreen! <br>

> For almost 5 years, the community patchers have been adding content to AM2R. Some of the creatures that didn't make it from the original Metroid II, new textures, X parasites, log books, and more have been added. This all works towards the final 2.0 update - a full graphical overhaul with unannounced content. To limit liability and respect the DMCA, we only distribute patches that are applied to the original AM2R 1.1 like a ROMhack. Our AM2RLauncher application automates this process for easy installation and updates.

Reviewer: All in all, not really liking these changes here.

Author: After rereading, I agree. Feel free to remove them.

Reviewer: Same as what I mentioned above.

Author: Removed

Author: Replaced with original

## Installation Instructions

#### How do I install the latest version on Windows?  
1. Head to the AM2RLauncher download page, download the Windows version, and unzip the file to its own folder: <https://github.com/AM2R-Community-Developers/AM2RLauncher/releases/latest>

> 2. Make sure that you have `.NET Framework 4.8` installed. The official download from Microsoft can be found here: <https://dotnet.microsoft.com/download/dotnet-framework/net48>

Reviewer: What's the point in removing the statement? Makes it sound like it has to be installed, which in 95% of cases doesn't have to be.

Author: People may skip the step if it says probably installed, I think having people download the installer just in case should be done.

Reviewer: > People may skip the step if it says probably installed

but thats the point. Again, in almost all cases it doesn't need to be done. And if it *does* need to be done, the launcher will actually pop up a window of its own (see question below in the faq somewhere)

Author: Good point, I did not realise there was a notice.

Author: reverted back

7. Click on the `Install` button.

#### How do I install the latest version on Linux?  

> 

Reviewer: Newline not necessary.

Author: Thanks for the catch, updated.

> For a universal installation of dependencies (except AM2R 1.1) and the AM2RLauncher on all Linux Distributions, including the Steam Deck, you have the option to install the AM2RLauncher via Flatpak:

Reviewer: Hm, not sure how I feel about just copy pasting the Wiki contents, as this means that every time something changes in the wiki (i.e., new dependency, new distro etc.) we'd have to paste it here for consistency.

Maybe it'd make more sense to have a structure more like this:
```
- mention to download am2rlauncher. mention flatpak/steam deck and that those can be used to skip further forwards into the instructions
- mention depdencies, a link to more detailed dependency install instructions in the wiki
- list out rest of the install instructions
```

Author: I mostly agree with this, however I feel the Flatpak instructions should stay, as it shows an easy way for the user to install the program, and only takes two lines (Steps 3-5 can be removed). I can redirect distribution specific commands to the wiki, but keep in mind if we list dependencies here then they will have to be updated as well. 

Here's what I have made:

To install on Linux with no dependencies on every distribution, you can use the Universal method with Flatpak, otherwise you can install the program manually with dependencies. 
**Universal/Steam Deck**
1. Follow the instructions for your Distro to install [Flatpak](https://flatpak.org/setup/) if don't have it installed already. Steam Deck users can skip this step.
2. Install the [AM2RLauncher Flatpak](https://flathub.org/apps/details/io.github.am2r_community_developers.AM2RLauncher). This may be from the linked site, or your software center, such as `Discover` on the Steam Deck.
3. In the Launcher, click on the `Download` button and let it download the necessary patch files.
4. Click on the `Select 1.1` button and select your copy of AM2R 1.1 and let the AM2RLauncher verify it.
5. Click on the `Install` button.

**Manual Install**
A distribution specific installation of AM2R is available on the [Community Updates Releases](https://github.com/AM2R-Community-Developers/AM2RLauncher/releases) page, and requires you to download the following dependencies:
`.NET Runtime` (5.0 or higher), `xdelta3`, `gtk3`, `libappindicator3`, `webkitgtk`, `openssl`, `fuse2`, Multilib / 32-Bit architecture added, and the **32-Bit version** of `libpulse`.
A detailed installation for each distribution with commands is available on the [GitHub Wiki](https://github.com/AM2R-Community-Developers/AM2RLauncher/wiki/Installation-Process) page.

Tried to keep it as thin as possible for maintenance and not confusing the user.

Reviewer: Ah sorry if i wasn't clear enough. I do want the flatpak mention to stay, just don't want to have a copy-pasted installation instruction. Something more akin to this
```
#### How do I install the latest version on Linux?  

###### Flatpak / Steam Deck
For a universal Linux install without the need to install any dependencies, or for Steam Deck users, you can choose to use the Flatpak version of the AM2RLauncher.
1. Follow the instructions for your Distro to install [Flatpak](https://flatpak.org/setup/) if don't have it installed already. Steam Deck users can skip this step.
2. Install the [AM2RLauncher Flatpak](https://flathub.org/apps/details/io.github.am2r_community_developers.AM2RLauncher). This may be from the linked site, or your software center, such as Discover on the Steam Deck.
3. Launch the AM2RLauncher, and follow the instructions from "Install Community Updates" further down.

First you need to download the right dependencies: `.NET Runtime` (6.0 or higher), `xdelta3`, `gtk3`, `libappindicator3`, `webkitgtk`, `openssl` and `fuse2`. In addition, you also need the following dependencies for AM2R itself: Multilib / 32-Bit architecture added, and the **32-Bit version** of `libpulse`.  
A detailed installation for each distribution with commands is available on the [GitHub Wiki](https://github.com/AM2R-Community-Developers/AM2RLauncher/wiki/Installation-Process) page.

After that, you need to download and start the AM2RLauncher:
1. Head to the AM2RLauncher download page, download the Linux version and unzip the file to its own folder: <https://github.com/AM2R-Community-Developers/AM2RLauncher/releases/latest>
2. Make sure that you have the following dependencies for the AM2RLauncher installed: 
3. Mark `AM2RLauncher.Gtk` as an executable, and then run it. This can be done like this: `chmod +x AM2RLauncher.Gtk` and `./AM2RLauncher`. Alternatively, if your desktop environment supports it, you can also double-click on the file after it has been marked as an executable.

###### Install Community Updates
1. Click on the `Download` button and let it download the necessary patch files.
2. Click on the `Select 1.1` button and select your copy of AM2R 1.1 and let the AM2RLauncher verify it.
3. Click on the `Install` button.
```

Author: Ah, my bad. I do think we should combine the flatpak and community updates install however. It being at the end may confuse some users who think the two middle paragraphs are needed for flatpak as well.

Reviewer: The same logic applies the other way around: people who install manually (which currently is our biggest userbase) being confused on what to do after installing dependencies.
I think that having them seperated is the better approach, as it may become useful in case we may get different downloading methods.

Yes. Depending on your internet speed and your download mirror's speed, this may take a while. You can try changing your mirror by going to the `Download Mirror` dropdown in the `Options` tab and seeing if that gets you a better speed.

#### I've encountered an issue with the AM2RLauncher. What do I do?  

> You can ask for support in the Official AM2R Discord/Matrix Server. Please provide relevant system information, steps you took, and the `AM2RLauncher.log` file which can be found in the `Logs` folder.  

Reviewer: Probably better to replace this with "communication channels", since it's not like we'll deny people who ask support via the subreddit or something.

Author: Sounds good, apply to all.

Author: Committed, resolving the other conversations related to this

#### I am on Linux and I don't want to use the AM2RLauncher. What do I do instead?  
This is not recommended, due to you missing out on automatic release updates, mod support, and easier use. However, the legacy installer can be found here:  
<https://github.com/AM2R-Community-Developers/AM2R-Autopatcher-Linux>  

> Again, it is not recommended to use the manual installation. For help, please use our GitHub issues page or the Official AM2R Discord/Matrix Server.

Reviewer: Same as above.

#### I am using Manjaro, and AM2R doesn't run. How do I fix this?  

> Unfortunately, we have been unable to figure out why AM2R will not run on Manjaro. Insights are greatly appreciated at our GitHub issues page or the Official AM2R Discord/Matrix Server.

Reviewer: Same as above.
And besides, I think this question can get removed now? From what I heard, am2r now runs on Manjaro but would need to get double checked.

Author: I dunno, I can check.

Reviewer: Would be thankful for it. Keep it in in this PR then, and if it does work on newer Manjaro releases, just make a seperate PR to remove it.

Author: Done, will check tonight

#### I am on Linux and I don't want to use the AM2RLauncher. What do I do instead?  
This is not recommended, due to you missing out on automatic release updates, mod support, and easier use. However, the legacy installer can be found here:  
<https://github.com/AM2R-Community-Developers/AM2R-Autopatcher-Linux>  
Again, it is not recommended to use the manual installation. For help, please use our GitHub issues page or communication channels.

#### I am using Manjaro, and AM2R doesn't run. How do I fix this?  

> Unfortunately, we have been unable to figure out why AM2R will not run on Manjaro. Insights are greatly appreciated at our GitHub issues page or communication channels.

Reviewer: *our

Author: done

#### I am on Linux and my controllers don't work!  
1. Check first if enabling `Compatibility Mode` in the Control options and restarting AM2R fixes the issue.
2. GameMaker: Studio 1.4 recommends the following dependencies, try to see if installing them fixes it: `jstest-gtk` and `joystick`.

> 3. If all fails, ask in the AM2R Discord/Matrix Server for more help.

Reviewer: See above.

#### I am on Linux and my controllers don't work!  
1. Check first if enabling `Compatibility Mode` in the Control options and restarting AM2R fixes the issue.
2. GameMaker: Studio 1.4 recommends the following dependencies, try to see if installing them fixes it: `jstest-gtk` and `joystick`.

> 3. If all fails, ask in communication channels for more help.

Reviewer: ditto.

#### I'm stuck in Random Game+! How do I proceed?  
Check first if you truly are stuck, you may have missed a path or an item somewhere.  

> Otherwise, post your **seed**, **game mode** and **operating system** into the Official AM2R Discord/Matrix Server in #am2r_help for people to help you!

Reviewer: See above.

#### I'm stuck in Random Game+! How do I proceed?  
Check first if you truly are stuck, you may have missed a path or an item somewhere.  

> Otherwise, post your **seed**, **game mode** and **operating system** into communication channels for people to help you!

Reviewer: ditto.

#### AM2R runs slow on Windows! Way slower than previous versions!  
YoYo Games made changes to some Windows-specific behavior that cause it to run slow on certain setups.  
To fix this, you may need to change the SleepMargin value in options.ini - higher values will take up more CPU cycles but will be less prone to the slowdown.  

> If you are confused about this issue, please ask in the Official AM2R Discord/Matrix Server for help.

Reviewer: See above.

#### AM2R runs slow on Windows! Way slower than previous versions!  
YoYo Games made changes to some Windows-specific behavior that cause it to run slow on certain setups.  
To fix this, you may need to change the SleepMargin value in options.ini - higher values will take up more CPU cycles but will be less prone to the slowdown.  

> If you are confused about this issue, please ask in communication channels for help.

Reviewer: ditto.

Otherwise, make sure that you have the 32-bit version of your drivers installed first.  
Then in the AM2RLauncher, go to the `Options` tab, and use the LD_PRELOAD environment variable that points to your OpenGL drivers.  
If you're on Ubuntu, this will probably look like this `LD_PRELOAD=/usr/lib/i386-linux-gnu/libGL.so.1`, if you're on Arch, it'll look like this `LD_PRELOAD=/usr/lib32/libGL.so`.  

> It may look different on other distros. If you have to use a different flag for a different distro, please let us know on the Official AM2R Discord/Matrix Server so we can include it here!

Reviewer: See above.

Otherwise, make sure that you have the 32-bit version of your drivers installed first.  
Then in the AM2RLauncher, go to the `Options` tab, and use the LD_PRELOAD environment variable that points to your OpenGL drivers.  
If you're on Ubuntu, this will probably look like this `LD_PRELOAD=/usr/lib/i386-linux-gnu/libGL.so.1`, if you're on Arch, it'll look like this `LD_PRELOAD=/usr/lib32/libGL.so`.  

> It may look different on other distros. If you have to use a different flag for a different distro, please let us know in communication channels so we can include it here!

Reviewer: ditto.

No, Multitroid is a mod made by MilesTheNerd. While we greatly enjoy the mod and Miles is part of our team, it is not officially supported at present due to the time and energy required to provide official multiplayer support.

#### How do I host on Multitroid?  

> 1. Install [AM2RServer](https://github.com/lassiterm/AM2R-Server/releases/latest)

Reviewer: A) having the link visible, makes it less likely that people are gonna miss to download it
B) use "download" instead of "install" please.

Author: My bad, I'll keep that in mind next time. I'll undo the link.

Author: Done

> 2. Port forward the server IP, or use a VPN (like Hamachi, ZeroTier, Radmin etc.).

Reviewer: ```suggestion
2. Either port forward, or use a VPN (like Hamachi, ZeroTier, Radmin etc.).
```

3. Launch the AM2R Server.
4. Give the other players your **external** IP address and port to connect with.

> 

Reviewer: This will break formatting. You probably want:
```
4. Give the other players your **external** IP address and port to connect with.  
    Port forwarding is different on every router, consult your routers manual or the internet for more info. For VPNs, consult their documentation. We cannot answer questions about specific routers or VPNs.
 ```

Author: Fixed

No, Multitroid is a mod made by MilesTheNerd. While we greatly enjoy the mod and Miles is part of our team, it is not officially supported at present due to the time and energy required to provide official multiplayer support.

#### How do I host on Multitroid?  

> 1. Download AM2RServer at <https://github.com/lassiterm/AM2R-Server/releases/latest>

Reviewer: ```suggestion
1. Download the corresponding AM2RServer version at <https://github.com/lassiterm/AM2R-Server/releases/latest>
```
Also, something I got reminded of: there's an unoffocial multitroid update version thing, with its own server. that maybe should get mentioned as well?

Author: I would suggest so, but I don't know the link or updater, I have not used multi yet.

Reviewer: Well, there are multiple. There is one version for horde, one for fair-team, one for unoficial updates and I think one or two more as well?
Was just food for thought, considering there are so many of them it may be better for them to get skipped.
Or just make a short notice that for other mods, their respective servers need to be used instead.

Author: Let's skip for now, I'll just commit your suggestion.
